src/db/options.rs

```diff
@@ -176,6 +177,35 @@ impl Default for ClusteredIndex {
     }
 }
 
+impl ClusteredIndex {
+    /// When creating a time-series collection on MongoDB Atlas the `clusteredIndex` field of the
+    /// collection options is given as `true` instead of as an object that deserializes to
+    /// `ClusteredIndex`. This custom deserializer handles that case by using the default value for
+    /// `ClusteredIndex`.
+    fn deserialize_self_or_true<'de, D>(deserializer: D) -> Result<Option<ClusteredIndex>, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        #[derive(Deserialize)]
+        #[serde(untagged)]
+        enum ValueUnion {
+            Bool(bool),
+            ClusteredIndex(ClusteredIndex),
+        }
+
+        let value_option: Option<ValueUnion> = Deserialize::deserialize(deserializer)?;
+        value_option
+            .map(|value| match value {
+                ValueUnion::Bool(value) if value == true => Ok(ClusteredIndex::default()),
+                ValueUnion::Bool(_) => Err(serde::de::Error::custom(
```

Comment:
Suggestion to fix clippy error:
```suggestion
                ValueUnion::Bool(true) => Ok(ClusteredIndex::default()),
                ValueUnion::Bool(false) => Err(serde::de::Error::custom(
```